File: backend/gene_model.py
from pydantic import BaseModel
from typing import Optional
from typing import List
from pydantic import BaseModel

class Gene(BaseModel):
    ensembl: str
    geneSymbol: Optional[str]
    name: Optional[str]
    bioType: str
    chromosome: str
    seqRegionStart: int
    seqRegionEnd: int

class GenesResponse(BaseModel):
    items: List[Gene]
    total: int
    page: int
    page_size: int
    pages: int

File: frontend/react-genes-app/src/models/gene.model.ts
import { BioType } from '@/enums/bio-type.enum';

export type Gene = {
  ensembl: string;
  geneSymbol: string;
  name: string;
  metadata: string;
  bioType: string;
  bioTypeCode: BioType;
  chromosome: string;
  seqRegionStart: number;
  seqRegionEnd: number;
};
